### ISSUE TITLE
another question :)

### ISSUE THREAD

**Issue by romanoss** [simon-larsson/keras-swa#6]
Hi, thx for the repo - very interesting!

I tried to use it in a kaggle competition, but till now I did not get it right.
In kaggle competitions one usually is finetuning with not a lot of epochs.
I call .fit() twice, first time eg 12 epochs with a warmup learning rate scheduler without swa and then again eg 17epochs with swa enabled.
Tried cyclic with freq 4, but val_loss got worse, not recovering from higher learning rates.
Tried cyclic with freq 8 and auto lr, but val_loss and val_auc are fluctuating and leading to sub-par results

Generally saved weights based on best val_loss do better in cv.

Can you please comment on this. Do you have any recommendations?

best
Roman

**Comment by simon-larsson** [comment]
First of all, SWA has no guarantees of always producing better results. If it does not improve results it does not mean you are using it wrong. Try it, and if it does not work you abandon it.

> Generally saved weights based on best val_loss do better in cv.

This does not work with SWA. The whole point of SWA is to replace the weights after training with an average that is thought to generalize better. If you then replace your weights with something else then you are no longer using SWA.

> Can you please comment on this. Do you have any recommendations?

Some suggestions:
- Rule of thumb is to use SWA for the last 25% of epochs (with warmup included). You control this with `start_epoch`.
- Use the constant schedule. It is preferred by the authors and cyclic only works when you have more epochs.

However, this is not an issue with the repo. So it is better you [email](mailto:larssonsimon0@gmail.com) me if you have more questions.
